### PR TITLE
BI-6174 Adding initial basic consumer

### PR DESCRIPTION
**Pull request by markpit**
Adding an initial consumer just to prove that we can read from kafka

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/processor/IncomingMessageConsumer.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Component
+public class IncomingMessageConsumer implements MessageConsumer {
+
+    @Value("${kafka.group.name}")
+    String groupName;
```
Comment (ICoxCH) [inline src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/processor/IncomingMessageConsumer.java]: I think these should be private really, and for unit testing we should use ReflectionTestUtils inside a BeforeEach annotated method to set these

Comment (mattch1) [reply src/main/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/processor/IncomingMessageConsumer.java]: Made fields private again. Kept init tests as it keeps the code coverage up.

File: src/test/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/processor/IncomingMessageConsumerUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    @Test
+    void testReadValidMessage() {
+        // TODO mock deserializer
+        List<Message> messages = new ArrayList<Message>();
+        when(consumer.consume()).thenReturn(messages);
+        Collection<ChipsKafkaMessage> result = incomingMessageConsumer.read();
+        assertTrue(result.isEmpty());
+    }
```
Comment (markpit) [inline src/test/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/processor/IncomingMessageConsumerUnitTest.java:88]: we should be able to test the error condition as well, the deserialise should throw an exception which we can use to test the catch block in read()

Comment (mattch1) [reply src/test/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/processor/IncomingMessageConsumerUnitTest.java:88]: The deserialise method is declared private.

**Review by mattch1 — APPROVED** [review]

**Pull request merged**